cpp/JSIHelper.cpp

```diff
@@ -143,40 +143,44 @@ jsi::Value createSequelQueryExecutionResult(jsi::Runtime &rt, SQLiteOPResult sta
   // Converting row results into objects
   size_t rowCount = results->size();
   jsi::Object rows = jsi::Object(rt);
-  if (rowCount > 0)
+  if (rowCount > 0 && metadata != NULL)
   {
     auto array = jsi::Array(rt, rowCount);
     for (int i = 0; i < rowCount; i++)
     {
       jsi::Object rowObject = jsi::Object(rt);
-      auto row = results->at(i);
-      for (auto const &entry : row)
+      auto row = results -> at(i);
+      // Iterate over metadata to maintain column order
+      for (const auto &column : *metadata)
       {
-        std::string columnName = entry.first;
-        QuickValue value = entry.second;
-        if (value.dataType == TEXT)
+        std::string columnName = column.colunmName;
```

Comment:
Should probably fix the typo in `colunmName`

Reply:
Option 2 sounds better to me, but based on what changes that would require in `bindings.cpp`/`sqliteExecute.cpp` I would like to defer that for now.